igvc_gazebo/nodes/ground_truth/main.cpp

```diff
@@ -36,8 +41,8 @@ void groundTruthCallback(const nav_msgs::Odometry::ConstPtr& msg)
     result.pose = msg->pose;
 
     // use the initial location as an offset (makes the starting location 0, 0)
-    result.pose.pose.position.x = msg->pose.pose.position.x - g_og_pose.pose.pose.position.x;
-    result.pose.pose.position.y = msg->pose.pose.position.y - g_og_pose.pose.pose.position.y;
+    result.pose.pose.position.x = msg->pose.pose.position.x - g_og_pose.pose.pose.position.x + x_distribution(engine);
+    result.pose.pose.position.y = msg->pose.pose.position.y - g_og_pose.pose.pose.position.y + y_distribution(engine);
```

Comment:
To expand on this, you could try adding in more fields (z, roll, pitch, yaw) to experiment and see which fields is our stack the most sensitive to.